src/createRollupFetchTransactionHash.ts

```diff
@@ -0,0 +1,47 @@
+import { deterministicFactoriesDeploymentEstimatedFees } from './constants';
+import { Address, PublicClient } from 'viem';
+import { AbiEvent } from 'abitype'
+
+const RollupInitializedEventAbi: AbiEvent = {
+  anonymous: false,
+  inputs: [
+      {
+          indexed: false,
+          internalType: "bytes32",
+          name: "machineHash",
+          type: "bytes32",
+      },
+      {
+          indexed: false,
+          internalType: "uint256",
+          name: "chainId",
+          type: "uint256",
+      },
+  ],
+  name: "RollupInitialized",
+  type: "event",
+};
+
+export async function createRollupFetchTransactionHash(
+  rollupAddress: Address,
+  publicClient: PublicClient,
+) {
+  // Find the RollupInitialized event from that Rollup contract
+  const rollupInitializedEvents = await publicClient.getLogs({
+    address: rollupAddress,
+    event: RollupInitializedEventAbi,
+    fromBlock: 'earliest',
+    toBlock: 'latest',
+  });
+  if (!rollupInitializedEvents || rollupInitializedEvents.length <= 0) {
+    throw new Error(`No RollupInitialized event found for rollup address ${rollupAddress}`);
+  }
```

Comment:
It doesn't seem like `rollupInitializedEvents` can ever be `undefined`, so I think we can just do this:
 
```suggestion
  if (rollupInitializedEvents.length !== 1) {
    throw new Error(`Expected to find 1 RollupInitialized event for rollup address ${rollupAddress} but found ${rollupInitializedEvents.length}`);
  }
```